client/landing/stepper/declarative-flow/internals/steps-repository/goals/select-goals.tsx

```diff
@@ -18,40 +19,54 @@ export const SelectGoals: React.FC< SelectGoalsProps > = ( {
 	const translate = useTranslate();
 	const goalOptions = useGoals();
 
-	const handleChange = ( selected: boolean, value: Onboard.SiteGoal ) => {
-		const newSelectedGoals = [ ...selectedGoals ];
-		const goalKey = value;
+	const handleChange = ( selected: boolean, goal: Onboard.SiteGoal ) => {
+		// Always remove potential duplicates
+		const newSelectedGoals = selectedGoals.filter( ( selectedGoal ) => selectedGoal !== goal );
```

Comment:
Hmm why should we have to do it? Having to add this in suggests we have some issue. If it is added for just to be safe, I do not think it is a good idea because it may cover up issues we may introduce.

Reply:
Okay so I'm running into duplicates from this import hackery, e.g. user pressing back from import page, and clicking import again, or clicking the submit button instead (and/or fiddle around the goal selections).

Need to find an approach that works when goals are updated outside of the component.

Reply:
Resolved by using Set.